draft_editor/draft_editor.py

```diff
@@ -0,0 +1,71 @@
+from talon import Module, actions, ui
+import time
+
+mod = Module()
+mod.mode("draft_editor", "Mode to show if the draft editor is open")
+
+active_window = None
+
+@mod.action_class
+class Actions:
+    def draft_editor_open():
+        """Open draft editor"""
+        global active_window
+        active_window = ui.active_window()
+        editor = get_editor()
+        has_selected_text = actions.edit.selected_text() != ""
+        if has_selected_text:
+            actions.edit.copy()
+        focus_window(editor)
+        new_file()
+        if has_selected_text:
+            actions.edit.paste()
+        actions.mode.enable("user.draft_editor")
+
+    def draft_editor_save():
+        """Save draft editor"""
+        close_editor(True)
+
+    def draft_editor_discard():
+        """Discard draft editor"""
+        close_editor(False)
+
+
+def close_editor(save: bool):
+    actions.mode.disable("user.draft_editor")
+    actions.edit.select_all()
+    if save:
+        actions.edit.copy()
+    close_file()
+    focus_window(active_window)
+    if save:
+        actions.edit.paste()
+
+def get_editor():
```

Comment:
I think I'd prefer to just use `actions.user.switcher_focus` here, because I believe this impl could grab a background VSCode window, instead of just switching to VSCode and using whatever is in front.  At least I have a recollection of this happening when I used this original version

Reply:
Is there such a thing as a background window? Applications yes, but windows?
If that is the case then background=False could probably be used.

Reply:
Sorry I don't mean background, but on a Mac, when you do your approach, it focused a random window, whereas if you just focus the app, it leaves the window focussed that was focussed previously

Reply:
Ah got it. No problem I can fix that.

Reply:
Should be fixed now. I use the latest window for the app.

Reply:
@AndreasArvidsson this still doesn't work for me.  It doesn't pick the most recent window.  Why not just use [`switcher_focus`](https://github.com/knausj85/knausj_talon/blob/b6f4939b3ce39785871c9e8ec88bbf91232c1090/code/switcher.py#L266-L276)?  That doesn't change the active window; just focuses the app, and is battle-tested across platforms.

Reply:
3 reasons
1. switcher_focus takes a signel app name. The draft editor needs to work for all different vscode names
2. switcher_focus doesn't wait for app to switch on other os then mac.
3. Switching back to the specific window is important.

It already works fine on windows and linux and I can update it to work better on mac but using switcher_focus is not a good fit for the above reasons.

Reply:
So I don't feel strongly enough to stand in the way of this PR; it is good to go from my perspective.  But I will say wrt each of your points above:

1. That should be an easy fix to `switcher_focus` that we might want to consider, no?
2. That's also an easy fix
3. I'm not sure how that would be affected by using `switcher_focus`, you could use the specific-window version here just like you do today

But as above, not a showstopper imo; happy to merge if @knausj85 is happy with this PR

Reply:
Great. If the appropriate functionality is available, or is made available, anywhere else in this repository I have no objections of using that instead.

Reply:
I have now made the necessary changes to the switcher so the draft editor can use those.